object_visualizer/src/object_visualizer_node.cpp

```diff
@@ -127,13 +135,33 @@ namespace object_visualizer
         marker.color.a = 1.0;
       }
 
+      int marker_size_x = obj.size.x;
+      int marker_size_y = obj.size.y;
+      int marker_size_z = obj.size.z;
+
+      marker.pose = obj.pose.pose;
+
+      // overwrite size in case any previous stack doesn't provide size
+      // such as carma_cooperative_perception at the moment
+      if (obj.size.x < 0.01 || obj.size.y < 0.01 || obj.size.z < 0.01)
+      {
+        marker_size_x = 1.0;
+        marker_size_y = 1.0;
+        marker_size_z = 1.0;
+        marker.pose.position.z = 1.0; // lifts the marker above ground so that it doesn't clip
```

Comment:
This can be made a little more robust:
```cpp
marker.pose.position.z = std::max(1.0, obj.size.z);
```
That way, the offset scales with the object size.

Reply:
added